src/libc/mod.rs

```diff
@@ -80,6 +83,86 @@ pub unsafe fn strlen(cs: *const c_char) -> size_t {
     count
 }
 
+pub struct EnvVal {
+    pub key: Cow<'static, [u8]>,
+    pub value: Cow<'static, [u8]>,
+}
+
+#[allow(unused)]
+static mut ENV_PTR: *const *const u8 = 0 as *const *const u8;
+static mut ENV: *mut HashMap<&'static [u8], EnvVal> = 0 as *mut _;
+#[allow(unused)]
+static mut AUXVAL: *const usize = 0 as *const usize;
+
+fn parse_env_str(input: &'static CStr) -> Option<(&'static [u8], &'static [u8])> {
+    let input = input.to_bytes();
+    if input.is_empty() {
+        return None;
+    }
+    let pos = memchr(b'=', &input[1..]).map(|p| p + 1);
```

Comment:
It wasn't obvious to me why `input` is sliced like this. Could you copy this comment from `std`?

```
        // Strategy (copied from glibc): Variable name and value are separated
        // by an ASCII equals sign '='. Since a variable name must not be
        // empty, allow variable names starting with an equals sign. Skip all
        // malformed lines.
```

I'm not sure if it's around already but if it is then all these `1..` slicing operations should point out to that comment.